tracer/src/Datadog.Trace/Iast/InsecureAuthAnalyzer.cs

```diff
@@ -0,0 +1,103 @@
+// <copyright file="InsecureAuthAnalyzer.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Linq;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Logging;
+#if !NETFRAMEWORK
+using Microsoft.AspNetCore.Http;
+using Microsoft.Net.Http.Headers;
+#endif
+
+namespace Datadog.Trace.Iast;
+
+#nullable enable
+
+internal static class InsecureAuthAnalyzer
+{
+    private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(InsecureAuthAnalyzer));
+
+#if !NETFRAMEWORK
+    public static void AnalyzeInsecureAuth(IHeaderDictionary headers, IntegrationId integrationId, int statusCode)
+    {
+        try
+        {
+            // Check for Authorization header
+            const string headerName = HeaderNames.Authorization;
+            if (!headers.TryGetValue(headerName, out var authHeader))
```

Comment:
authHeader is a StringValues that you are implicitly converting later to string. By default, the StringValues class will join the values with a "," when calling ToString() if there are more than one value. Have you considered this case?

Reply:
Thanks for pointing this out!
You're right, I totally forgot that this could be a possibility. I added another step in the parsing step and some tests so all cases can work.